plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetFileWriterFactory.java

```diff
@@ -127,6 +145,44 @@ public Optional<FileWriter> createFileWriter(
                     HIVE_PARQUET_USE_LEGACY_DECIMAL_ENCODING,
                     HIVE_PARQUET_USE_INT96_TIMESTAMP_ENCODING);
 
+            Optional<Function<ParquetWriteValidation, ConnectorPageSource>> validationInputFactory = Optional.empty();
+            if (isParquetOptimizedWriterValidate(session)) {
+                validationInputFactory = Optional.of(writeValidation -> {
+                    try {
+                        long fileSize = fileSystem.getFileStatus(path).getLen();
+                        List<HiveColumnHandle> columns = IntStream.range(0, fileColumnNames.size()).boxed()
+                                .map(index -> new HiveColumnHandle(
+                                        fileColumnNames.get(index),
+                                        index,
+                                        hiveTypes.get(index),
+                                        fileColumnTypes.get(index),
+                                        Optional.empty(),
+                                        HiveColumnHandle.ColumnType.REGULAR,
+                                        Optional.empty()))
+                                .collect(toImmutableList());
+                        return ParquetPageSourceFactory.createPageSource(
+                                path,
+                                0L,
+                                fileSize,
+                                fileSize,
+                                columns,
+                                TupleDomain.all(),
+                                true,
+                                hdfsEnvironment,
+                                hdfsEnvironment.getConfiguration(new HdfsEnvironment.HdfsContext(session), path),
+                                session.getIdentity(),
+                                parquetTimeZone,
+                                readStats,
+                                new ParquetReaderOptions(),
+                                Optional.of(writeValidation))
```

Comment:
How it's solved in ORC? Could we just create
```
                        return new HdfsOrcDataSource(
                                new OrcDataSourceId(path.toString()),
                                fileSystem.getFileStatus(path).getLen(),
                                new OrcReaderOptions(),
                                fileSystem.open(path),
                                readStats);
```
as in orc?

This reader creation code seems pretty complicated